src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java

```diff
@@ -177,14 +185,21 @@ public void setDashboards(final OpenshiftRequestContext context, String kibanaVe
      * with it
      */
     private List<Project> filterProjectsWithIndices(Set<Project> projects) {
-        List<Project> result = new ArrayList<>(projects.size());
+        List<String> patterns = new ArrayList<>(projects.size());
         for (Project project : projects) {
             String indexPattern = kibanaUtils.formatIndexPattern(project);
-            if (pluginClient.indexExists(indexPattern)) {
-                result.add(project);
-            }
+            patterns.add(indexPattern);
+        }
+        LOGGER.trace("Evaluating {} indexPattern for existing index.", patterns.size());
+        Set<Project> result = new HashSet<>(projects.size());
+        SearchResponse response = pluginClient.search(patterns.toArray(new String[]{}), ArrayUtils.EMPTY_STRING_ARRAY, 1000, false);
+        for (SearchHit hit : response.getHits().getHits()){
```

Comment:
can `response.getHits()` return null?  If so, need to guard against NPE

Reply:
done